gitlab/resource_gitlab_instance_cluster_test.go

```diff
@@ -0,0 +1,301 @@
+package gitlab
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabInstanceCluster_basic(t *testing.T) {
+	var cluster gitlab.InstanceCluster
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabInstanceClusterDestroy,
+		Steps: []resource.TestStep{
+			// Create an instance cluster with default options
+			{
+				Config: testAccGitlabInstanceClusterConfig(rInt, true),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabInstanceClusterExists("gitlab_instance_cluster.foo", &cluster),
+					testAccCheckGitlabInstanceClusterAttributes(&cluster, &testAccGitlabInstanceClusterExpectedAttributes{
+						Name:                        fmt.Sprintf("foo-cluster-%d", rInt),
+						Domain:                      "example.com",
+						EnvironmentScope:            "*",
+						KubernetesApiURL:            "https://123.123.123",
+						KubernetesCACert:            instanceClusterFakeCert,
+						KubernetesAuthorizationType: "abac",
+					}),
+				),
+			},
+			// create an unmanaged cluster
+			{
+				Config: testAccGitlabInstanceClusterConfig(rInt, false),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabInstanceClusterExists("gitlab_instance_cluster.foo", &cluster),
+					testAccCheckGitlabInstanceClusterAttributes(&cluster, &testAccGitlabInstanceClusterExpectedAttributes{
```

Comment:
This test step is a bit odd because your `Config` is different yet your `Check` is identical to the prior step. Can you update your `testAccGitlabInstanceClusterExpectedAttributes` helper to assert that the cluster changed from managed to unmanaged?